tools/nightly_build_and_test_stats.groovy

```diff
@@ -325,6 +359,86 @@ def verifyReleaseContent(String version, String release, String variant, Map sta
     }
 }
 
+// For a given pipeline, tell us how reproducible the builds were.
+// Note: Will limit itself to jdk versions and platforms in the results Map.
+def getReproducibilityPercentage(String jdkVersion, String trssId, String trssURL, Map results) {
+    echo "Called repro method with trssID:"+trssId
+
+    def platformConversionMap = getPlatformConversionMap()
+    def platformReproTestMap = getPlatformReproTestMap()
+
+    // We are only looking for reproducible percentages for the relevant jdk versions...
+    if ( trssId != "" && results.containsKey(jdkVersion) ) {
+        def jdkVersionInt = jdkVersion.replaceAll("[a-z]", "")
+
+        // ...and platforms.
+        results[jdkVersion][1].each { onePlatform, valueNotUsed ->
+            // If this platform doesn't have a reproducibility test yet, skip it.
+            if (platformReproTestMap[onePlatform][0].equals("NA")) {
+                results[jdkVersion][1][onePlatform] = "NA"
+                // Then we exit this lambda and skip to the next platform.
+                return
+            }
+
+            def pipelineLink = trssURL+"/api/getAllChildBuilds?parentId="+trssId+"\\&buildNameRegex=^"+jdkVersion+"\\-"+platformConversionMap[onePlatform][0]+"\\-temurin\$"
+            def trssBuildJobNames = sh(returnStdout: true, script: "wget -q -O - ${pipelineLink}")
+            def platformResult = "???% - Build not found. Pipeline link: " + pipelineLink
+
+            // Does this platform have a build in this pipeline?
+            if ( trssBuildJobNames.length() > 10 ) {
```

Comment:
Not quite sure of the significance of "10" here?

Reply:
Hi Andrew. The significance of 10 was me noticing that even if the wget returns an empty array, the square brackets still give us a string length of 2. On the first draft, I didn't know if there was a space in the middle (there isn't) so I just went with 10 on the basis that a returned string with a non-empty json would have dozens of characters.

I've changed this to 2.